frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import { useSize } from './drawer-utils';
+import './Drawer.scss';
+
+type DrawerProps = {
+  height?: number;
+  defaultHeight?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+  maxHeight?: number | string;
+  isResizable?: boolean;
+  header: React.ReactNode;
+  headerClassName?: string;
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  defaultHeight,
+  height,
+  maxHeight = '100%',
+  defaultOpen,
+  open = defaultOpen,
+  isResizable = false,
+  header,
+  headerClassName,
+  onChange,
+}) => {
+  const [heightState, setHeightState] = React.useState(defaultHeight);
+  const [lastObservedHeight, setLastObservedHeight] = React.useState<number>(null);
+  const [size, ref] = useSize<HTMLDivElement>();
+  const resizeHeight = height || heightState;
+  const minimumHeight = size.height || 0;
+  const handleDrag = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (newHeight < 0) return;
+    if (onChange) {
+      onChange(open, newHeight);
+    } else {
+      setHeightState(newHeight);
+    }
+  };
+
+  const handleResizeStart = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (newHeight >= minimumHeight) onChange(true, newHeight);
+    setLastObservedHeight(newHeight);
+  };
+  const handleResizeStop = (e: MouseEvent, data: DraggableData) => {
```

Comment:
I don't think you need this if you do everything in start and drag functions.

Reply:
If we don't do this We will be setting the lastObservedHeight on drag instead when the DragStopped

Reply:
Ya. Doesn't that make sense? Why wouldn't you be setting proper height and open state on drag when the height becomes less than min height? Why wait till they stop dragging?

Reply:
Nope, Consider this - you start dragging down from 300 and once you reach the minHeight you set it to lastObservedHeight i.e. 300, Instead of stopping the drag you continue the drag upwards now, parent component already have the height as lastobservedHeight then on the next callback  newHeight is `300 - ${delta}`. This will cause the fluctuation in the height.
![fluc](https://user-images.githubusercontent.com/9278015/76729689-bdbcdc80-677f-11ea-8519-ba794d9fd5d2.gif)

Reply:
Ok that's fine. I was thinking of having more smarts in this component to know when to render min height vs the expected height. This will work though.